flink-connectors/flink-connector-hbase/src/test/java/org/apache/flink/connector/hbase/testutil/HBaseTestCluster.java

```diff
@@ -0,0 +1,328 @@
+    }
+
+    public void shutdownCluster() {
+        LOG.info("Shutting down HBase test cluster");
+        try {
+            try (Closer closer = Closer.create()) {
+                closer.register(this::clearTables);
+                closer.register(this::clearReplicationPeers);
+            }
+            try {
+                // Closer is not able to call this method correctly, instead logs process dump
+                cluster.shutdown();
+            } catch (IOException e) {
+                LOG.error("Error while shutting down HBase test cluster", e);
+            }
+            try (Closer closer = Closer.create()) {
+                closer.register(this::waitForShutDown);
+                closer.register(Paths.get(testFolder).toFile()::delete);
+            }
+        } catch (IOException e) {
+            throw new RuntimeException(
+                    "Failed to shut down HBase test cluster. Future program state might be compromised.",
+                    e);
+        }
+        LOG.info("HBase test cluster shut down");
+    }
+
+    public void waitForShutDown() {
+        try {
+            CompletableFuture.runAsync(cluster::waitUntilShutDown).get(240, TimeUnit.SECONDS);
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            LOG.error("Interrupted while waiting for HBase test cluster to shut down", e);
+        } catch (ExecutionException e) {
+            e.printStackTrace();
+            LOG.error("Exception while waiting for HBase test cluster to shut down", e);
+        } catch (TimeoutException e) {
+            e.printStackTrace();
+            LOG.error("Waiting for HBase test cluster to shut down timed out", e);
+        }
+    }
+
+    public boolean canConnectToCluster() throws InterruptedException, ExecutionException {
+        try {
+            return CompletableFuture.supplyAsync(
+                            () -> {
+                                try (Connection connection =
+                                        ConnectionFactory.createConnection(getConfig())) {
+                                    return true;
+                                } catch (IOException e) {
+                                    LOG.error("Error trying to connect to cluster", e);
+                                    return false;
+                                }
+                            })
+                    .get(10, TimeUnit.SECONDS);
+        } catch (TimeoutException e) {
+            LOG.error("Trying to connect to HBase test cluster timed out", e);
+            return false;
+        }
+    }
+
+    public void clearTables() {
+        try (Admin admin = ConnectionFactory.createConnection(getConfig()).getAdmin()) {
+            for (TableDescriptor table : admin.listTableDescriptors()) {
+                admin.disableTable(table.getTableName());
+                admin.deleteTable(table.getTableName());
+            }
+        } catch (IOException e) {
```

Comment:
You can let `IOExceptions` just bubble up to the `Closer`. If you want to disable/close as many tables as possible, you should add them individually to closer (or create a new one here).

Reply:
✅ in b0e0f5d7a0bc332b81aa949d357de063f7a00d98